### PR TITLE
Support mso_site for ND hosted MSO

### LINKED ISSUE #40: mso_site not compatible with Nexus Dashboard hosted versions of MSO (v3.3 and v3.4)
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.0.4
on darwin_amd64
+ provider registry.terraform.io/ciscodevnet/mso v0.2.1

### MSO version

Any Nexus Dashboard based version. Tested with MSO 3.3(1e) and 3.4(1a) on Nexus Dashboard version 2.0(2g)
(I use the Cisco dCloud lab "Cisco Nexus Dashboard Orchestrator for ACI Lab v1") 

### APIC version and APIC Platform for Site Level Resources

* V 5.1(1h) on-prem

### Affected Resource(s)

* mso_site

### Terraform Configuration Files

Used the same configuration as the site example in this repo, but with "nd" as platform.
https://github.com/CiscoDevNet/terraform-provider-mso/blob/master/examples/site/main.tf

```hcl
terraform {
  required_providers {
    mso = {
      source = "CiscoDevNet/mso"
      version = "0.2.1"
    }
  }
}

provider "mso" {
    username = "admin"
    password = "C1sco12345"
    url      = "https://198.18.133.100/"
    insecure = true
    platform = "nd"
}

resource "mso_site" "site_test_1" {
  name             = "Cisco_MSO se"
  username         = "admin"
  password         = "noir0!234"
  apic_site_id     = "18"
  urls             = ["https://3.208.123.222"]
  # login_domain     = "radius_test"
  # maintenance_mode = true
  location = {
    lat  = 78.946
    long = 95.623
  }
}

```

### Debug Output

https://gist.github.com/andreassb/3cac87448daa3676021cb09bce887878

### Panic Output

https://gist.github.com/andreassb/5d2275213572db5fbb56ae43f29a56ba

### Expected Behavior

Site created in Nexus Dashboard, and prefferably marked as "managed" in MSO.

### Actual Behavior

No sites created.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

MSO 3.3 and 3.4 handles handles sites differently than the previous versions of MSO, because it is hosted on Nexus Dashboard.

### PR DESCRIPTION
**Pull request by xinyuezhao**
resolved #40 
- Add function GetSiteViaName() for mso_site resource
- Support mso_site for ND hosted MSO

### CODE REVIEW

File: mso/resource_mso_site.go [hunk 1]
```diff
@@ -168,56 +169,94 @@ func resourceMSOSiteCreate(d *schema.ResourceData, m interface{}) error {
...
-		loc_map := location.(map[string]interface{})
-		loc.Lat, _ = strconv.ParseFloat(fmt.Sprintf("%v", loc_map["lat"]), 64)
-		loc.Long, _ = strconv.ParseFloat(fmt.Sprintf("%v", loc_map["long"]), 64)
-		siteAttr.Location = loc
-	}
-
-	if urls, ok := d.GetOk("urls"); ok {
-		siteAttr.Url = urls.([]interface{})
-	}
+	var apiVersion string
```
Comment (shrsr) [inline mso/resource_mso_site.go:176]: Why is apiVersion declared 4 times?

File: mso/resource_mso_site.go [hunk 2]
```diff
@@ -168,56 +169,94 @@ func resourceMSOSiteCreate(d *schema.ResourceData, m interface{}) error {
...
+	var path string
```
Comment (shrsr) [inline mso/resource_mso_site.go:177]: Why has path has been declared thrice?

**Review by lhercot — CHANGES_REQUESTED** [review]

File: mso/resource_mso_site.go (hunk 1 above)
Comment (xinyuezhao) [reply mso/resource_mso_site.go:176]: we declare these variables like msoClient, siteAttr in each resourceMSOSiteCreate/resourceMSOSiteUpdate/resourceMSOSiteRead/resourceMSOSiteDelete func. I suppose they are not duplicate

File: mso/resource_mso_site.go (hunk 2 above)
Comment (xinyuezhao) [reply mso/resource_mso_site.go:177]: same as above

**Review by lhercot — DISMISSED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**